UnitTests/OIDTVAuthorizationRequestTests.m

```diff
@@ -0,0 +1,198 @@
+    
+    [bodyParameters setObject:components[1] forKey:components[0]];
+  }
+
+  return bodyParameters;
+}
+
+- (void)testInitializer {
+  OIDTVAuthorizationRequest *authRequest =
+      [[OIDTVAuthorizationRequest alloc] initWithConfiguration:[self testServiceConfiguration]
+                                                      clientId:kTestClientID
+                                                  clientSecret:kTestClientSecret
+                                                        scopes:nil
+                                          additionalParameters:nil];
+
+  XCTAssertEqualObjects(authRequest.responseType, OIDResponseTypeCode);
+  XCTAssertEqualObjects(authRequest.redirectURL, [[NSURL alloc] init]);
+}
+
+- (void)testURLRequestBasicClientAuth {
+  OIDTVAuthorizationRequest *authRequest =
+      [[OIDTVAuthorizationRequest alloc] initWithConfiguration:[self testServiceConfiguration]
+                                                      clientId:kTestClientID
+                                                  clientSecret:kTestClientSecret
+                                                        scopes:nil
+                                          additionalParameters:nil];
+
+  NSURLRequest *urlRequest = [authRequest URLRequest];
+
+  XCTAssertEqualObjects([urlRequest HTTPMethod], kHTTPPost);
+  XCTAssertEqualObjects([urlRequest valueForHTTPHeaderField:kHTTPContentTypeHeaderKey],
+                        kHTTPContentTypeHeaderValue);
+  XCTAssertEqualObjects(urlRequest.URL.absoluteString, kInitializerTestTVAuthEndpoint);
+
+  NSDictionary<NSString *, NSString *> *bodyParameters =
+      [self bodyParametersFromURLRequest:urlRequest];
+  NSDictionary<NSString *, NSString *> *expectedParameters = @{@"client_id" : kTestClientID};
+
+  XCTAssertEqualObjects(bodyParameters, expectedParameters);
+}
+
+- (void)testURLRequestScopes {
+  OIDTVAuthorizationRequest *authRequest =
+      [[OIDTVAuthorizationRequest alloc] initWithConfiguration:[self testServiceConfiguration]
+                                                      clientId:kTestClientID
+                                                  clientSecret:kTestClientSecret
+                                                        scopes:@[ kTestScope, kTestScopeA ]
+                                          additionalParameters:nil];
+
+  NSURLRequest *urlRequest = [authRequest URLRequest];
+
+  XCTAssertEqualObjects([urlRequest HTTPMethod], kHTTPPost);
+  XCTAssertEqualObjects([urlRequest valueForHTTPHeaderField:kHTTPContentTypeHeaderKey],
+                        kHTTPContentTypeHeaderValue);
+  XCTAssertEqualObjects(urlRequest.URL.absoluteString, kInitializerTestTVAuthEndpoint);
+
+  NSDictionary<NSString *, NSString *> *bodyParameters =
+      [self bodyParametersFromURLRequest:urlRequest];
+  NSDictionary<NSString *, NSString *> *expectedParameters = @{
+    @"client_id" : kTestClientID,
+    @"scope" : [[NSString stringWithFormat:@"%@ %@", kTestScope, kTestScopeA]
```

Comment:
nit: It would be nice to have @"client_id" and @"scope" defined as static consts (or more correctly expose the consts in OIDAuthorizationRequest.m). We can leave that for later.